src/provider-mysql.tf

```
@@ -9,7 +9,7 @@ locals {
 
   mysql_admin_user         = module.aurora_mysql.outputs.aurora_mysql_master_username
   mysql_admin_password_key = module.aurora_mysql.outputs.aurora_mysql_master_password_ssm_key
-  mysql_admin_password     = local.enabled ? (length(var.mysql_admin_password) > 0 ? var.mysql_admin_password : data.aws_ssm_parameter.mysql_admin_password[0].value) : ""
+  mysql_admin_password     = local.enabled ? (length(var.mysql_admin_password) > 0 ? var.mysql_admin_password : data.aws_ssm_parameter.admin_password[0].value) : ""
```

```suggestion
  mysql_admin_password     = local.enabled ? (length(var.mysql_admin_password) > 0 ? var.mysql_admin_password : one(data.aws_ssm_parameter.admin_password[*].value)) : ""
```

Thank you for this fix! Mind if we add some additional validation to this improvement as well?

---

Sure, no problem.